### PR TITLE
CORE-16716 tenant id validation generates clearer error message

### PR DESCRIPTION
**Pull request by bpaunescu**

### CODE REVIEW

File: components/membership/membership-rest-impl/src/main/kotlin/net/corda/membership/impl/rest/v1/HsmRestResourceImpl.kt [hunk 1]
```diff
@@ -125,8 +126,13 @@ class HsmRestResourceImpl @Activate constructor(
...
-            tenantId
-        ) { "Could not find holding identity '$tenantId' associated with member." }
+        try {
+            virtualNodeInfoReadService.getByHoldingIdentityShortHashOrThrow(
+                tenantId
+            ) { "Could not find holding identity '$tenantId' associated with member." }
+        } catch (e: BadRequestException) {
+            // re-throw using a more suitable message
+            throw BadRequestException("Invalid tenant id. Accepted values are holding identity short hashes, 'p2p', or 'rest'")
+        }
```
Comment (anien) [inline components/membership/membership-rest-impl/src/main/kotlin/net/corda/membership/impl/rest/v1/HsmRestResourceImpl.kt]:
```suggestion
            throw BadRequestException("Invalid tenant id. Accepted values are holding identity short hash, 'p2p', or 'rest'.")
        }
```

File: components/membership/membership-rest-impl/src/main/kotlin/net/corda/membership/impl/rest/v1/HsmRestResourceImpl.kt [hunk 2]
```diff
@@ -125,8 +126,13 @@ class HsmRestResourceImpl @Activate constructor(
...
-        virtualNodeInfoReadService.getByHoldingIdentityShortHashOrThrow(
-            tenantId
-        ) { "Could not find holding identity '$tenantId' associated with member." }
+        try {
+            virtualNodeInfoReadService.getByHoldingIdentityShortHashOrThrow(
+                tenantId
+            ) { "Could not find holding identity '$tenantId' associated with member." }
+        } catch (e: BadRequestException) {
+            // re-throw using a more suitable message
+            throw BadRequestException("Invalid tenant id. Accepted values are holding identity short hashes, 'p2p', or 'rest'")
```
Comment (yift-r3) [inline components/membership/membership-rest-impl/src/main/kotlin/net/corda/membership/impl/rest/v1/HsmRestResourceImpl.kt]: Shouldn't we put the request `tenantId` in the message?

Comment (bpaunescu) [reply components/membership/membership-rest-impl/src/main/kotlin/net/corda/membership/impl/rest/v1/HsmRestResourceImpl.kt]: Have added the tennatId to the message

File: components/membership/membership-rest-impl/src/main/kotlin/net/corda/membership/impl/rest/v1/HsmRestResourceImpl.kt (hunk 1 above)
Comment (bpaunescu) [reply components/membership/membership-rest-impl/src/main/kotlin/net/corda/membership/impl/rest/v1/HsmRestResourceImpl.kt]: Updated, thanks!

**Review by yift-r3 — APPROVED** [review]
LGTM

**Pull request merged**